src/irc/ConnectionInstance.ts

```diff
@@ -424,7 +424,7 @@ export class ConnectionInstance {
 
                 // Closing Link: gateway/shell/matrix.org/session (Bad user info)
                 // ircd-seven doc link: https://git.io/JvxEs
-                if ((err.args[0] as string|null)?.match(/Closing Link: .+\(Bad user info\)/)) {
+                if (err.args && err.args[0]?.match(/Closing Link: .+\(Bad user info\)/)) {
```

Comment:
Why the question mark on `err.args[0]?`? You can't run `.match` on `undefined`.
Should we also test if `err.args[0]` is actually defined?

Reply:
That's what `?` does :). It's a Typescript feature that returns null early if `args` is not truthy. Hence, err.args[0].match will not be evaluated if [0] is undefined.

Reply:
In that case, why is this not `err.args?[0]?.match(…`?